src/main/java/com/hbm/tileentity/machine/TileEntityBroadcaster.java

```diff
@@ -1,55 +1,103 @@
 package com.hbm.tileentity.machine;
 
 import java.util.List;
+import java.util.Random;
 
 import com.hbm.lib.ModDamageSource;
-import com.hbm.packet.PacketDispatcher;
-import com.hbm.packet.toclient.LoopedSoundPacket;
+import com.hbm.main.MainRegistry;
 
-import cpw.mods.fml.common.network.NetworkRegistry.TargetPoint;
+import com.hbm.sound.AudioWrapper;
+import com.hbm.tileentity.TileEntityLoadedBase;
 import cpw.mods.fml.relauncher.Side;
 import cpw.mods.fml.relauncher.SideOnly;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.entity.EntityClientPlayerMP;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.potion.Potion;
 import net.minecraft.potion.PotionEffect;
 import net.minecraft.tileentity.TileEntity;
 import net.minecraft.util.AxisAlignedBB;
 
-public class TileEntityBroadcaster extends TileEntity {
-	
+public class TileEntityBroadcaster extends TileEntityLoadedBase {
+
+	private AudioWrapper audio;
+
 	@Override
 	public void updateEntity() {
-		
+
 		List<Entity> list = worldObj.getEntitiesWithinAABBExcludingEntity(null, AxisAlignedBB.getBoundingBox(xCoord + 0.5 - 25, yCoord + 0.5 - 25, zCoord + 0.5 - 25, xCoord + 0.5 + 25, yCoord + 0.5 + 25, zCoord + 0.5 + 25));
-		
+
 		for(int i = 0; i < list.size(); i++) {
 			if(list.get(i) instanceof EntityLivingBase) {
 				EntityLivingBase e = (EntityLivingBase)list.get(i);
 				double d = Math.sqrt(Math.pow(e.posX - (xCoord + 0.5), 2) + Math.pow(e.posY - (yCoord + 0.5), 2) + Math.pow(e.posZ - (zCoord + 0.5), 2));
-				
+
 				if(d <= 25) {
 					if(e.getActivePotionEffect(Potion.confusion) == null || e.getActivePotionEffect(Potion.confusion).getDuration() < 100)
 						e.addPotionEffect(new PotionEffect(Potion.confusion.id, 300, 0));
 				}
-				
+
 				if(d <= 15) {
 					double t = (15 - d) / 15 * 10;
 					e.attackEntityFrom(ModDamageSource.broadcast, (float) t);
 				}
 			}
 		}
 
-		if(!worldObj.isRemote) {
-			PacketDispatcher.wrapper.sendToAllAround(new LoopedSoundPacket(xCoord, yCoord, zCoord), new TargetPoint(worldObj.provider.dimensionId, xCoord, yCoord, zCoord, 150));
+		if (worldObj.isRemote) {
+			if(audio == null) {
+				audio = createAudioLoop();
+				audio.startSound();
+			} else if(!audio.isPlaying()) {
+				audio = rebootAudio(audio);
+			}
+
+			int intendedVolume = 25;
+
+			EntityClientPlayerMP player = Minecraft.getMinecraft().thePlayer;
```

Comment:
please use `MainRegistry.proxy.me()` for this, `Minecraft.getMinecraft()` and any other client-only calls like that are a big nono even in `world.isRemote` blocks